### PR TITLE
[8.0][FIX][account_move_batch_validate] Filter on journals doesn't work

### PR DESCRIPTION
**Pull request by adrienpeiffer**
Currently, changing journals in the wizard has no effect. All journals are posted.

### CODE REVIEW

File: account_move_batch_validate/wizard/move_marker.py [hunk 1]
```diff
@@ -38,11 +38,16 @@ class AccountMoveMarker(orm.TransientModel):
             ('mark', 'Mark for posting'),
             ('unmark', 'Unmark for posting'),
         ], "Action", required=True),
-        'eta': fields.integer('Seconds to wait before starting the jobs')
+        'eta': fields.integer('Seconds to wait before starting the jobs'),
+        'journal_ids': fields.many2many('account.journal',
+                                        'account_post_journal_rel',
+                                        'wiz_id', 'journal_id', 'Journals',
+                                        required=True),
```
Comment (pedrobaeza) [inline account_move_batch_validate/wizard/move_marker.py]: I prefer to let it not required, and only filter if there is any value here.

Comment (adrienpeiffer) [inline account_move_batch_validate/wizard/move_marker.py]: Hi @pedrobaeza ! 
I think It doesn't really make sense to not define any journals here.
What do you think ?

Comment (sbidoul) [inline account_move_batch_validate/wizard/move_marker.py]: In the official wizard validate.account.move the list of period and journals is initially empty and the user must select explicitly the one he/she wants to post.

In this batch validate module, it proposes the list of existing journals and the user must remove the one he/she does not want to validate.

I personally prefer the behaviour of the official module.

Comment (pedrobaeza) [inline account_move_batch_validate/wizard/move_marker.py]: Well, it's a matter of taste, but I prefer to not having anything, and add only the desired ones when needed, than removing everything except one, that it's the common task. I mean, my usual tasks are:
- Validate all journals.
- Validate only 1 journal.

In these use cases, I prefer my approach.

Comment (adrienpeiffer) [inline account_move_batch_validate/wizard/move_marker.py]: According with Stéphane and concidering that this module replace the official menu for draft entries validation, I'll propose a solution to reajust this one from the official wizard.

**Comment by adrienpeiffer** [comment]
@sbidoul @pedrobaeza The last commit contains a solution to reajust the asynchronous wizard from the official one

**Comment by sbidoul** [comment]
install_in_connector() is deprecated. When you remove it you should get a green runbot.

File: account_move_batch_validate/wizard/move_marker_view.xml [hunk 2]
```diff
@@ -2,58 +2,22 @@
...
-                    <group col="4">
-                        <field name="chart_account_id" widget='selection' on_change="onchange_chart_id(chart_account_id, context)"/>
-                        <field name="company_id" invisible="1"/>
-                        <field name="fiscalyear_id" domain="[('company_id','=',company_id)]"/>
-                        <field name="action"/>
+                <xpath expr="//group" position="before">
+                    <group>
+                        <field name="asynchronous" />
+                        <field name="action" attrs="{'invisible': [('asynchronous', '=', False)]}"/>
                     </group>
```
Comment (sbidoul) [inline account_move_batch_validate/wizard/move_marker_view.xml]: Can you try without creating a new group for asynchronous and action? The look should be better.

**Comment by adrienpeiffer** [comment]
@sbidoul I rebase this one to get https://github.com/OCA/account-financial-tools/commit/709b90c57fcdf5add878427571299239401359d1

**Comment by sbidoul** [comment]
This way the behaviour of the filter will match the upstream wizard.

:+1:

**Comment by yvaucher** [comment]
@adrienpeiffer Just a little update of translations and then :+1:

**Comment by adrienpeiffer** [comment]
@yvaucher It's done ! Thanks !

File: account_move_batch_validate/i18n/fr.po [hunk 3]
```diff
@@ -1,192 +1,79 @@
...
+msgid "UUID of the Job to approve this move"
+msgstr "UUID du Job pour approuver cette move"
 
 #. module: account_move_batch_validate
-#: help:account.move.marker,chart_account_id:0
-msgid "Select Charts of Accounts"
-msgstr "Sélectionner Plan Comptable"
+#: selection:validate.account.move,action:0
+msgid "Unmark for posting"
+msgstr "Déchocher pour la validation"
```
Comment (yvaucher) [inline account_move_batch_validate/i18n/fr.po]: s/Déchocher/Décocher

File: account_move_batch_validate/i18n/fr.po [hunk 4]
```diff
@@ -1,192 +1,79 @@
...
+msgid "Check this box to mark the move for batch posting"
+msgstr "Cocher cette case pour marquer les écritures pour une validation asynchrone"
 
 #. module: account_move_batch_validate
-#: view:account.move.marker:0
-msgid "Mark"
-msgstr "Mark"
+#: selection:validate.account.move,action:0
+msgid "Mark for posting"
+msgstr "Marquer pour poster"
```
Comment (yvaucher) [inline account_move_batch_validate/i18n/fr.po]: s/pour poster/pour validation ?

File: account_move_batch_validate/i18n/fr.po [hunk 5]
```diff
@@ -1,192 +1,79 @@
...
+msgid "Task set to Done because the user unmarked the move"
+msgstr "Tâche mise à terminé car l'utilisateur a décoché la pièce"
 
 #. module: account_move_batch_validate
-#: field:account.move.marker,date_from:0
-msgid "Start Date"
-msgstr "Date de début"
+#: field:account.move,post_job_uuid:0
+msgid "UUID of the Job to approve this move"
+msgstr "UUID du Job pour approuver cette move"
```
Comment (yvaucher) [inline account_move_batch_validate/i18n/fr.po]: s/move/pièce comptable

File: account_move_batch_validate/i18n/fr.po [hunk 6]
```diff
@@ -1,192 +1,79 @@
...
+msgid "Use asynchronous validation"
+msgstr "Utilisation de la validation asynchrone"
 
 #. module: account_move_batch_validate
-#: view:account.move.marker:0
-msgid "or"
-msgstr "ou"
+#: model:ir.model,name:account_move_batch_validate.model_validate_account_move
+msgid "Validate Account Move"
+msgstr "Valider les mouvements de compte"
```
Comment (yvaucher) [inline account_move_batch_validate/i18n/fr.po]: s/mouvements de compte/pièces comptables

**Comment by yvaucher** [comment]
Some translations to fix

For coherence with https://github.com/odoo/odoo/blob/ad253a33863c0e2b7bf4add5f689461120278a1b/addons/account/i18n/fr.po#L708

Please use same naming for account move

**Comment by adrienpeiffer** [comment]
Thanks @yvaucher. Translations are up to date.

**Comment by yvaucher** [comment]
@adrienpeiffer Thanks :+1:

**Comment by gurneyalex** [comment]
@adrienpeiffer thanks! :smile_cat: 

:+1:

**Pull request merged**